### DISCUSSION TITLE
Sofle Pico ver 3.5.4 Cannot drag and drop Flash Using Current UF2

### DISCUSSION THREAD

**Discussion by kaide** [JellyTitan/Sofle-Pico#30] [Q&A] [answered]
Hi Ryan, I started buying material for Sofle Pico last February (PCB ver 3.5.4 ), just got time to do soldering this month. 

I want to flash and test just with 1 switch connected for just the left hand PCB.

The problem is I cannot drag and drop flash using current UF2 (left hand) downloaded from https://github.com/JellyTitan/Sofle-Pico/tree/main/Sofle_Pico/Firmware . When the UF2 file is copied,  the PICO did not restart. 

When I did drag and drop flash using blink.uf2 from https://www.raspberrypi.com/documentation/microcontrollers/c_sdk.html#blink-an-led the PICO restart and blink.

I tried using the Sofle Pico main branch revision ebc424c646666acf03161d9c9283167d423d3d07 (Fixes title logo dark-mode variant on mobile) and the PICO also restart after dropping the UF2 file. The encoder and key works. But when connecting to VIA site, the layout are not visible.

My RP2040 is the 3 buttons one with 264KB SRAM and 4MB flash.
This is the content of my Pico's INFO_UF2.txt:
UF2 Bootloader v3.0
Model: Raspberry Pi RP2
Board-ID: RPI-RP2

So is this a bug for the current UF2? Or should I just use the firmware for 3.5.4 version?

Thanks

**Comment by JellyTitan** [comment]
Interesting! 

A few questions:
1. Are you on a PC or a mac?
2. Which vendor did you buy the Pico from?

**Reply by JellyTitan** [reply]
It's also worth noting - the firmware is stuck in limbo at the moment - there's an open QMK PR, which blocks a VIA PR & the main soflekeyboard PR. 

The latest QMK branch had to be heavily modified because QMK is focusing on merging only the bare minimum required for a board - which ment pulling out a bunch of functionality.  (Also switched from RH/LH .uf2 files to a single file for both RH/LH) There may be a mismatch between via capabilities and the QMK branch.

I was able to get VIA to recognize the board using these files:
https://github.com/JellyTitan/Sofle-Pico/blob/main/Sofle_Pico/Firmware/sofle_pico_default_via_RH.uf2
https://github.com/JellyTitan/Sofle-Pico/blob/main/Sofle_Pico/Firmware/sofle_pico_default_via_LH.uf2

![image](https://github.com/JellyTitan/Sofle-Pico/assets/72168556/4aba1aac-d9d3-486b-9cb0-0d13afb9a3a1)


Were those the files that were giving you trouble?

**Comment by kaide** [comment, marked as answer]
Ah its my fault, I copied the UF2 file the wrong way, the one I got has html element. Invalid and bigger size :)

I use the download button, from the https://www.soflepico.com/flashing/drag-and-drop and https://github.com/JellyTitan/Sofle-Pico/tree/main/Sofle_Pico/Firmware, same as your link above.

Now it works.

Sorry for the trouble.

Thanks

**Reply by JellyTitan** [reply]
No worries! Glad it worked for you!
